browser/resources/settings/brave_overrides/basic_page.ts

```diff
@@ -308,6 +309,20 @@ RegisterPolymerTemplateModifications({
           prefs: '{{prefs}}'
         }
       ))
+      const sectionLeoCustomModels = document.createElement('template')
+      sectionLeoCustomModels.setAttribute('is', 'dom-if')
+      sectionLeoCustomModels.setAttribute('restamp', 'true')
+      sectionLeoCustomModels
+        .setAttribute('if', '[[showPage_(pageVisibility.leoCustomModels)]]')
+      sectionLeoCustomModels.content.appendChild(createNestedSectionElement(
+        'leoAssistant',
+        'leoAssistant',
+        'braveLeoAssistantByomLabel',
+        'model-list-section',
+        {
+          prefs: '{{prefs}}'
+        }
+      ))
```

Comment:
This should all be contained within the existing leo page, not sure why we need a second page when the url doesn't even change

Reply:
leaving it as it is because maybe we might want this as a route in the future.

Reply:
well it's not because of the separate route, but because of the separate sections on the same page and in both chromium and brave they are all defined in basic_page. I do think we don't need these helper functions though in order to make nested sections, but they're already there.